app/models/vpago/payment_decorator.rb

```diff
@@ -7,6 +7,22 @@ def self.prepended(base)
       base.after_update :cancel_pre_auth, if: :state_changed_to_failed?
     end
 
+    # @override
+    # order trigger payment.process! when calling order.next.
+    def process!
+      return complete! if payment_method.auto_capture? && payment_receive_manually?
+
```

Comment:
Let's remove the reference to ticket_seller_user?. This gem is designed to be abstract and should not have any knowledge of the existence of ticket_seller_user within the order. Keeping the gem decoupled ensures it remains flexible and reusable across different business domains without being tied to specific implementations.

```
def payment_received_manually?
    check_payment_method? && payment_received?
end

# Override this method in your specific business use-case to define how payment confirmation is handled.
# For example, in a walk-in sale process, you can implement the logic to verify how payment is received.
def payment_received?
    false
end
```

Reply:
I understand it completely, that's why I put it called business logics inside `payment_receive_manually?`

We can't put it inside commissioner because all `ticket seller` logics are here in spree_vpago not in spree_cm_commissioner. We need to migrate the whole thing to do so.

Reply:
I also use `recieve` instead of `recieved` because I was thinking of following distinction when doing them.

```rb
def payment_receive_manually?
    order.ticket_seller_user?
end

def payment_received_manually?
    state == 'complete' && order.ticket_seller_user?
end
```